src/pages/resultsView/ResultsViewURLWrapper.ts

```diff
@@ -314,33 +317,70 @@ export default class ResultsViewURLWrapper
         }
     }
 
-    public getOncoprintClinicalTrackParams(clinicalTracks: string[]) {
-        let clinicallist: string;
-        if (clinicalTracks.length > 0) {
-            clinicallist = clinicalTracks.join(',');
+    /**
+     * Query param clinicallist can be:
+     * - url encoded json object
+     * - comma seperated list (legacy)
+     * - undefined: clinical tracks not configured
+     * - 'null': all clinical tracks were deleted by user
+     *
+     * Ideally, we would like to simply give an empty string.
+     * The problem is that, in order to know whether to show
+     * some clinical tracks by default (such as "profiled-in",
+     * "samples per patient", etc), we need to know whether
+     * the clinical tracks have been updated by the user. If we
+     * pass an empty string, the router just deletes that parameter
+     * from the URL completely, making it indistinguishable from
+     * the initialization state. So we have to use 'null' here to
+     * distinguish the state of user having deleted all clinical tracks,
+     * because the alternative is to make a breaking change to the router library.
+     *
+     * @param clinicalTracks
+     */
+    public convertClinicalTracksToUrlParam(
+        clinicalTracks: ClinicalTrackConfig[]
+    ) {
+        if (clinicalTracks.length) {
+            return { clinicallist: JSON.stringify(clinicalTracks) };
         } else {
-            // ideally, we would like to simply give an empty string.
-            //   The problem is that, in order to know whether to show
-            //   some clinical tracks by default (such as "profiled-in",
-            //   "samples per patient", etc), we need to know whether
-            //  the clinical tracks have been updated by the user. If we
-            //  pass an empty string, the router just deletes that parameter
-            //  from the URL completely, making it indistinguishable from
-            //  the initialization state. So we have to use "null" here to
-            //  distinguish the state of user having deleted all clinical tracks,
-            //  because the alternative is to make a breaking change to the router library.
-
-            clinicallist = 'null';
+            return { clinicallist: ALL_TRACKS_DELETED };
```

Comment:
could we just use an empty array instead of null?

Reply:
For backward compatibility (and possible bookmarks) I guess we should keep the legacy functionality as is